nixos/modules/system/vars/options.nix

```diff
@@ -0,0 +1,185 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+{
+  options.vars = {
+    settings = {
+      fileModule = lib.mkOption {
+        type = lib.types.deferredModule;
+        internal = true;
+        description = ''
+          A module to be imported in every vars.files.<name> submodule.
+          Used by backends to define the `path` attribute.
+
+          Takes the file as an argument and returns maybe an attrset which should at least contain the `path` attribute.
+          Can be used to set other file attributes as well, like `value`.
+        '';
+        default = { };
+      };
+    };
+    generators = lib.mkOption {
```

Comment:
This contains both generators (executables), as well as their target files (calls of those executables).
Since those are in a 1:many relationship (potentially), this part of the option tree should be named after the "target files" instead (or whichever term those are described with).

Reply:
It seems that you have an insofar unnamed domain entity for groups of files that are generated together, and perhaps its name should be put here.

Reply:
hmm, the idea was, that one generator can create many files, so the files are below the generators in the hierarchy. Files belong to a generator in that they need to be generated by one, not sure where else in the option tree they could live

Reply:
I agree with @roberth. Files are very similar to services in the sense of [modular services](https://github.com/NixOS/nixpkgs/pull/372170): their collection could be a canonical option in the NixOS configuration tree, and it doesn't matter how they are produced as long as their producers implement an agreed-upon interfaces.